### ISSUE TITLE
IDE0059 (Unnecessary assignment of a value) should not be reported for variables that are passed by reference

### ISSUE THREAD

**Issue by LeroyK** [dotnet/roslyn#40717]
**Version Used**: 
VS2019 16.4.2
.NET Core 3.1

**Steps to Reproduce**:
```c#
static int Example()
{
    int value = 123;

    Span<int> valueSpan = MemoryMarshal.CreateSpan(ref value, 1);

    value = 456;

    return valueSpan[0];
}
```

**Expected Behavior**:
No suggestion to remove the assignment ``value = 456;``

**Actual Behavior**:
Suggestion to remove ``value = 456;``

**Comment by jinujoseph** [comment]
Related to #37709
